src/compiler/parser.ts

```diff
@@ -1397,7 +1397,14 @@ namespace ts {
             // Only for end of file because the error gets reported incorrectly on embedded script tags.
             const reportAtCurrentPosition = token() === SyntaxKind.EndOfFileToken;
 
-            return createMissingNode<Identifier>(SyntaxKind.Identifier, reportAtCurrentPosition, diagnosticMessage || Diagnostics.Identifier_expected);
+            const isReservedWord = tokenIsKeyword(token());
+            const msgArg = scanner.getTokenText();
+
+            const baseMessage = isReservedWord ?

```

Comment:
I think you might be able to use `scanner.isReservedWord()` instead of adding the `tokenIsKeyword` function.

Reply:
Thanks! I've made the change.